selftest/selftest.py

```diff
@@ -0,0 +1,869 @@
+        else:
+            assert (
+                self.disk_info.scsi_resource_disk is None
+                and self.disk_info.dev_disk_azure_resource_disk is None
+            ), f"unexpected resource disk path: {self.disk_info}"
+
+        logger.info(
+            "validate_scsi_resource_disk OK: /dev/disk/azure/resource => %s",
+            self.disk_info.dev_disk_azure_resource_disk,
+        )
+
+    def validate_sku_config(self) -> None:
+        """Validate SKU config."""
+        if not self.sku_config:
+            logger.warning(
+                "validate_sku_config SKIPPED: no sku configuration for VM size %r",
+                self.vm_size,
+            )
+            return
+
+        assert (
+            self.sku_config.vm_size == self.vm_size
+        ), f"vm size mismatch: {self.sku_config.vm_size} != {self.vm_size}"
+        assert (
+            len(self.disk_info.nvme_local_disks)
+            == self.sku_config.nvme_local_disk_count
+        ), f"local disk count mismatch: {len(self.disk_info.nvme_local_disks)} != {self.sku_config.nvme_local_disk_count}"
+        assert (
+            self.disk_info.nvme_local_disk_size_gib
+            == self.sku_config.nvme_local_disk_size_gib
+        ), f"local disk size mismatch: {self.disk_info.nvme_local_disk_size_gib} != {self.sku_config.nvme_local_disk_size_gib}"
+        assert (
+            self.disk_info.scsi_resource_disk_size_gib
+            == self.sku_config.temp_disk_size_gib
+        ), f"temp disk size mismatch: {self.disk_info.scsi_resource_disk_size_gib} != {self.sku_config.temp_disk_size_gib}"
+        assert (
+            self.disk_info.dev_disk_azure_resource_disk_size_gib
+            == self.sku_config.temp_disk_size_gib
+        ), f"temp disk size mismatch: {self.disk_info.dev_disk_azure_resource_disk_size_gib} != {self.sku_config.temp_disk_size_gib}"
+
+        logger.info("validate_sku_config OK: %r", self.sku_config)
+
+    def validate(self) -> None:
+        """Run validations."""
+        self.azure_nvme_id_info.validate(self.disk_info)
+        self.validate_dev_disk_azure_links_data()
+        self.validate_dev_disk_azure_links_local()
+        self.validate_dev_disk_azure_links_os()
+        self.validate_dev_disk_azure_links_resource()
+        self.validate_scsi_resource_disk()
+        self.validate_sku_config()
+        logger.info("success!")
+
+
+def main() -> None:
+    """Main entry point."""
+    parser = argparse.ArgumentParser(
+        description="Azure VM utilities self-tests script."
+    )
+    parser.add_argument(
+        "--debug",
+        action="store_true",
+        help="Enable debug logging",
+    )
+    args = parser.parse_args()
+
+    if args.debug:
+        logging.basicConfig(format="%(message)s", level=logging.DEBUG)
+    else:
+        logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+    validator = AzureVmUtilsValidator()
+    validator.validate()
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Does most of the logic from this script comes from tuxops?

Reply:
Yeah, it largely matches, and in cases exceeds the validation work done there.